### PR TITLE
DiracDelta(-x) == DiracDelta(x)

### PR DESCRIPTION
**Pull request by smichr**
* Minimal simplification to DiracDelta is done when the argument presents an explicit literal, negative coefficient. More complicated expressions will respond to `signsimp` as in the added tests.

* signsimp will now pass along the `evaluate` flag during its recursive step.

### CODE REVIEW

File: sympy/functions/special/delta_functions.py [hunk 1]
```diff
@@ -193,8 +193,16 @@ def eval(cls, arg, k=0):
             return S.NaN
         if arg.is_nonzero:
             return S.Zero
+        c, nc = arg.args_cnc()
+        if c and c[0] == -1:
+            # keep this fast and simple instead of using
+            # could_extract_minus_sign
+            return cls(-arg, k) if k else cls(-arg)
```
Comment (jksuom) [inline sympy/functions/special/delta_functions.py]: The derivative of order `k` has the same parity as `k`. The first derivative of an even function is odd, the second derivative is even, etc.

File: sympy/functions/special/delta_functions.py [hunk 2]
```diff
@@ -194,7 +197,18 @@ def eval(cls, arg, k=0):
...
+                Complex part: %s  found in %s .''' % (
+                repr(im(arg)), repr(arg))))
+        c, nc = arg.args_cnc()
+        if c and c[0] == -1:
+            # keep this fast and simple instead of using
+            # could_extract_minus_sign
+            if k % 2 == 1:
+                return -cls(-arg, k)
+            elif k % 2 == 0:
+                return cls(-arg)
```
Comment (jksuom) [inline sympy/functions/special/delta_functions.py]: I believe `k` should be here as well.

Comment (smichr) [reply sympy/functions/special/delta_functions.py]: Thanks! I added a test and updated the function.

File: sympy/functions/special/delta_functions.py [hunk 3]
```diff
@@ -40,16 +40,19 @@ class DiracDelta(Function):
...
-    4) ``DiracDelta(g(x)) = Sum_i(DiracDelta(x-x_i)/abs(g'(x_i)))``
+    4) ``DiracDelta(g(x)) = Sum_i(DiracDelta(x - x_i)/abs(g'(x_i)))``
        Where ``x_i``-s are the roots of ``g``
+    5) ``DiracDelta(-x) = DiracDelta(x)``
 
     Derivatives of ``k``-th order of DiracDelta have the following property:
 
-    5) ``DiracDelta(x,k) = 0``, for all ``x != 0``
+    6) ``DiracDelta(x, k) = 0``, for all ``x != 0``
+    7) ``DiracDelta(-x, k) = -DiracDelta(x, k)`` for odd ``k``
```
Comment (smichr) [inline sympy/functions/special/delta_functions.py]: @jksuom, have I captured the right behavior here in 7 and 8?

**Comment by smichr** [comment]
Does anyone have an opinion as to whether `could_extract_minus_sign` should be used instead of jest checking for a leading factor of -1?

File: sympy/functions/special/delta_functions.py (hunk 3 above)
Comment (jksuom) [reply sympy/functions/special/delta_functions.py]: Yes, I think so. These are distributions, not functions, but they obey the same rules as functions.

**Pull request merged**